packages/smooth_app/lib/pages/history_page.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_list_page.dart';
+
+class HistoryUpdatedSemaphore extends ChangeNotifier {
+  bool needsReload = true;
+  // Notify the history page that it needs to reload.
+  void notify() {
+    needsReload = true;
+    super.notifyListeners();
+  }
+}
+
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
+  @override
+  State<HistoryPage> createState() => _HistoryPageState();
+}
+
+class _HistoryPageState extends State<HistoryPage> {
+  late DaoProductList _daoProductList;
+  ProductList productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    final HistoryUpdatedSemaphore semaphore =
+        context.watch<HistoryUpdatedSemaphore>();
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
```

Comment:
Moved the discussion to Slack, since I feel it might be good to get more input into this: https://openfoodfacts.slack.com/archives/CSV2748EB/p1636976728008500

Reply:
Using `didChangeDependencies` as discussed.